### ISSUE TITLE
Plain python version of dump_data (for a PyTorch port)

### ISSUE THREAD

**Issue by snakers4** [mozilla/LPCNet#85]
Hi @jmvalin! 

Many thanks for your amazing work. We are authors of [open_stt](https://github.com/snakers4/open_stt) and [open_tts](https://github.com/snakers4/open_tts) (large open Russian datasets). We are planning to release ~12k speakers and ~15k hours worth of TTS data for the Russian language (also soon we will be writing a piece on how to make STT networks 10x faster w/o sacrificing performance).

We achieved some relative success in training TTS networks with Tacotron2 + WaveGlow (which can by the way run 5-10 real-time-speed on one GPU). 

But this is nevertheless too slow, even compared with our STT network, which already runs 20-30x real-time-speed on a 10-core CPU (this maybe relatively slow, but we do not write in C and such speed is good enough for practical applications) without any non-neural-network optimizations (e.g. PyTorch has compilers, you can try to run the network in half accuracy, etc etc).

We looked at your keras python code, and so far it looks like that [porting](https://github.com/mozilla/LPCNet/blob/master/src/train_lpcnet.py) it to PyTorch is quite doable.

From what I saw in this [thread](https://github.com/mozilla/LPCNet/issues/4), the quality for the Russian language using the default neural network leaves much to be desired. Also this [thread](https://github.com/mozilla/LPCNet/issues/52) suggests that just slapping Tacotron above your `dump_data` also works if you are careful with adding noise.

So, I would like to ask, if there is a python version of `dump_data`? 
If you could help us with this, this would enable it to integrate pre-processing inside of PyTorch's `dataloader` abstractions + add custom augmentations on top. Usually for projects with a lot of data pre-computing features may be quite burdensome (for example for STT we use stft done directly in PyTorch).

Many thanks!
